core/src/main/java/io/questdb/cairo/wal/seq/SequencerMetadata.java

```diff
@@ -0,0 +1,314 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal.seq;
+
+import io.questdb.cairo.*;
+import io.questdb.cairo.sql.TableRecordMetadata;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryMARW;
+import io.questdb.cairo.vm.api.MemoryMR;
+import io.questdb.std.*;
+import io.questdb.std.str.Path;
+
+import java.io.Closeable;
+
+import static io.questdb.cairo.TableUtils.*;
+import static io.questdb.cairo.wal.WalUtils.*;
+
+public class SequencerMetadata extends BaseRecordMetadata implements TableRecordMetadata, Closeable, TableDescriptor {
+    private final FilesFacade ff;
+    private final boolean readonly;
+    private final MemoryMARW metaMem;
+    private final MemoryMR roMetaMem;
+
+    private long structureVersion = -1;
+    private int tableId;
+    private String tableName;
+    private boolean suspended;
+
+    public SequencerMetadata(FilesFacade ff) {
+        this(ff, false);
+    }
+
+    public SequencerMetadata(FilesFacade ff, boolean readonly) {
+        this.ff = ff;
+        this.readonly = readonly;
+        if (!readonly) {
+            roMetaMem = metaMem = Vm.getMARWInstance();
```

Comment:
nit: probably it makes sense to use `MemoryMR` for `roMetaMem` in `!readonly` case.

Reply:
i think `roMetaMem` is using the `MemoryMR` interface already.
or is this about something else?

Reply:
Yes, my bad. Let me resolve this one.